cuFFT callback tests fail on sm_86

On the latest master + sm_86, the failure happens at the linking stage.

Reproducer:
```
$ pytest tests/cupy_tests/fft_tests/test_callback.py -x
========================================================================= test session starts =========================================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/leofang/dev/cupy_cuda114, configfile: setup.cfg
collected 584 items                                                                                                                                                   

tests/cupy_tests/fft_tests/test_callback.py F

============================================================================== FAILURES ===============================================================================
_____________________________________________ Test1dCallbacks.test_fft_load[_param_0_{n=None, norm=None, shape=(10, 7)}] ______________________________________________

self = <<cupy_tests.fft_tests.test_callback.Test1dCallbacks object at 0x7f470e296a60>  parameter: {'n': None, 'norm': None, 'shape': (10, 7)}>
xp = <module 'cupy' from '/home/leofang/dev/cupy_cuda114/cupy/__init__.py'>, dtype = <class 'numpy.complex64'>

    @testing.for_complex_dtypes()
    @testing.numpy_cupy_allclose(rtol=1e-4, atol=1e-7, contiguous_check=False)
    def test_fft_load(self, xp, dtype):
>       return self._test_load_helper(xp, dtype, 'fft')

tests/cupy_tests/fft_tests/test_callback.py:146: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/cupy_tests/fft_tests/test_callback.py:138: in _test_load_helper
    with xp.fft.config.set_cufft_callbacks(cb_load=cb_load):
cupy/fft/_callback.pyx:567: in cupy.fft._callback.set_cufft_callbacks.__init__
    mgr = _CallbackManager(
cupy/fft/_callback.pyx:388: in cupy.fft._callback._CallbackManager.__init__
    _nvcc_link(tempdir, obj_host, obj_dev, arch, mod_filename,
cupy/fft/_callback.pyx:290: in cupy.fft._callback._nvcc_link
    p.check_returncode()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CompletedProcess(args=['/usr/local/cuda-11.4/bin/nvcc', '-ccbin', 'g++', '-shared', '-arch=sm_86', '/tmp/tmpj_icd884/t...mpj_icd884/tmp8i5gc_9o/cupy_callback_a84d83fce0114cf19d18471ad9a57fd5.cpython-38-x86_64-linux-gnu.so'], returncode=255)

    def check_returncode(self):
        """Raise CalledProcessError if the exit code is non-zero."""
        if self.returncode:
>           raise CalledProcessError(self.returncode, self.args, self.stdout,
                                     self.stderr)
E           AssertionError: Only cupy raises error
E           
E           Traceback (most recent call last):
E             File "/home/leofang/dev/cupy_cuda114/cupy/testing/_helper.py", line 47, in _call_func
E               result = impl(self, *args, **kw)
E             File "/home/leofang/dev/cupy_cuda114/tests/cupy_tests/fft_tests/test_callback.py", line 146, in test_fft_load
E               return self._test_load_helper(xp, dtype, 'fft')
E             File "/home/leofang/dev/cupy_cuda114/tests/cupy_tests/fft_tests/test_callback.py", line 138, in _test_load_helper
E               with xp.fft.config.set_cufft_callbacks(cb_load=cb_load):
E             File "cupy/fft/_callback.pyx", line 567, in cupy.fft._callback.set_cufft_callbacks.__init__
E               mgr = _CallbackManager(
E             File "cupy/fft/_callback.pyx", line 388, in cupy.fft._callback._CallbackManager.__init__
E               _nvcc_link(tempdir, obj_host, obj_dev, arch, mod_filename,
E             File "cupy/fft/_callback.pyx", line 290, in cupy.fft._callback._nvcc_link
E               p.check_returncode()
E             File "/home/leofang/miniconda3/envs/cupy_cuda114_dev/lib/python3.8/subprocess.py", line 448, in check_returncode
E               raise CalledProcessError(self.returncode, self.args, self.stdout,
E           subprocess.CalledProcessError: Command '['/usr/local/cuda-11.4/bin/nvcc', '-ccbin', 'g++', '-shared', '-arch=sm_86', '/tmp/tmpj_icd884/tmp8i5gc_9o/cupy_callback_a84d83fce0114cf19d18471ad9a57fd5_dev.o', '/tmp/tmpj_icd884/tmp8i5gc_9o/cupy_callback_a84d83fce0114cf19d18471ad9a57fd5.o', '-L/tmp/tmpj_icd884', '-lcufft_static_10500_sm86', '-lculibos', '-lpthread', '-o', '/tmp/tmpj_icd884/tmp8i5gc_9o/cupy_callback_a84d83fce0114cf19d18471ad9a57fd5.cpython-38-x86_64-linux-gnu.so']' returned non-zero exit status 255.

../../miniconda3/envs/cupy_cuda114_dev/lib/python3.8/subprocess.py:448: AssertionError
------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------
dtype is <class 'numpy.complex64'>
------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------
nvlink error   : Undefined reference to 'lut_dp_2_13' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_17' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_19' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_23' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_29' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_31' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_37' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_41' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_43' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_47' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_53' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_59' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_61' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_67' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_71' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_73' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_79' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_83' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_89' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_97' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_101' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_103' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_107' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_109' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_113' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_127' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_8_32' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_8_64' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_8_256' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_8_512' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_16_64' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_16_128' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_16_256' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_16_1024' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_16_2048' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_16_4096' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_2_4' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_3_9' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_4_8' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_4_16' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_5_25' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_5_125' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_5_625' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_5_3125' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_6_36' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_6_216' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_6_1296' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_7_49' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_7_343' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_7_2401' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_8_2048' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_8_4096' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_9_27' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_9_81' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_9_243' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_9_729' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_9_2187' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_10_100' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_10_1000' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_11_121' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_11_1331' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_12_144' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_12_1728' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_32_256' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_dp_32_8192' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbdouble_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_13' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_17' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_19' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_23' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_29' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_31' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_37' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_41' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_43' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_47' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_53' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_59' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_61' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_67' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_71' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_73' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_79' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_83' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_89' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_97' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_101' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_103' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_107' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_109' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_113' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_127' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_prime_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_8_32' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_8_64' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_8_256' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_8_512' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_32' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_64' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_128' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_256' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_512' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_1024' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_2048' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_4096' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_16_8192' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_bluestein_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_2_4' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_3_9' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_4_8' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_4_16' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_5_25' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_5_125' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_5_625' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_5_3125' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_6_36' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_6_216' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_6_1296' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_7_49' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_7_343' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_7_2401' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_9_27' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_9_81' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_9_243' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_9_729' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_9_2187' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_10_100' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_10_1000' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_11_121' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_11_1331' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_12_144' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
nvlink error   : Undefined reference to 'lut_sp_12_1728' in '/tmp/tmpj_icd884/libcufft_static_10500_sm86.a:cbfloat_32bit_regular_callback_RT_PTX.o'
======================================================================= short test summary info =======================================================================
FAILED tests/cupy_tests/fft_tests/test_callback.py::Test1dCallbacks::test_fft_load[_param_0_{n=None, norm=None, shape=(10, 7)}] - AssertionError: Only cupy raises e...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================================================== 1 failed in 5.69s ==========================================================================
```

Environment:
```python
>>> cp.show_config()
OS                           : Linux-5.4.0-72-generic-x86_64-with-glibc2.10
Python Version               : 3.8.10
CuPy Version                 : 10.0.0a2
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.21.0
SciPy Version                : 1.7.0
Cython Build Version         : 0.29.24
Cython Runtime Version       : 0.29.24
CUDA Root                    : /usr/local/cuda-11.4
nvcc PATH                    : /usr/local/cuda-11.4/bin/nvcc
CUDA Build Version           : 11040
CUDA Driver Version          : 11040
CUDA Runtime Version         : 11040
cuBLAS Version               : (available)
cuFFT Version                : 10500
cuRAND Version               : 10205
cuSOLVER Version             : (11, 2, 0)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 4)
Thrust Version               : 101201
CUB Build Version            : 101201
Jitify Build Version         : 60e9e72
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : 2507
NCCL Runtime Version         : 2507
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA RTX A6000
Device 0 Compute Capability  : 86
Device 0 PCI Bus ID          : 0000:4B:00.0
```

--- a/cupy/fft/_callback.pyx
+++ b/cupy/fft/_callback.pyx
@@ -39,7 +39,6 @@ cdef str _cuda_include = None
 cdef str _nvprune = None
 cdef str _build_ver = None
 cdef int _cufft_ver = 0
-cdef list _cufft_static_no_device_code = []
 cdef str _cupy_root = None
 cdef str _cupy_include = None
 cdef str _source_dir = None
@@ -194,28 +193,45 @@ cdef inline void _mod_compile(str tempdir, str mod_name, str obj_host) except*:
     p.check_returncode()
 
 
+# Archs with the same major CC version could share the implementations, so for
+# the purpose of pruning we should dump all SMs with the same major ver.
+#
+# In practice, this only happens with sm_86 (cupy/cupy#5508) as of CUDA 11.4;
+# other SMs always use sm_X0's symbols only and do not have their own
+# specializations.
+cdef dict _cc_major_map = {
+    '8': ('80', '86'),
+    '7': ('70', '72', '75'),
+    '6': ('60', '61', '62'),
+    '5': ('50', '52', '53'),
+    '3': ('35', '37'),
+}
+
+
 cdef inline str _prune(str temp_dir, str cache_dir, str _cufft_ver, str arch):
     cdef str cufft_lib_full, cufft_lib_pruned, cufft_lib_temp, cufft_lib_cached
 
-    if _nvprune and arch not in _cufft_static_no_device_code:
+    if _nvprune:
         cufft_lib_full = os.path.join(_cuda_path, 'lib64/libcufft_static.a')
-        cufft_lib_pruned = 'cufft_static_' + _cufft_ver + '_sm' + arch
+        cufft_lib_pruned = f'cufft_static_{_cufft_ver}_sm{arch[0]}'
         cufft_lib_temp = os.path.join(temp_dir,
                                       'lib' + cufft_lib_pruned + '.a')
         cufft_lib_cached = os.path.join(cache_dir,
                                         'lib' + cufft_lib_pruned + '.a')
         if not os.path.isfile(cufft_lib_cached):
-            p = subprocess.run([_nvprune, '-arch=sm_' + arch,
-                                cufft_lib_full, '-o', cufft_lib_temp],
+            comm = [_nvprune]
+            for cc in _cc_major_map[arch[0]]:
+                comm.append(f'--generate-code=arch=compute_{cc},code=sm_{cc}')
+            comm += [cufft_lib_full, '-o', cufft_lib_temp]
+            p = subprocess.run(comm,
                                env=os.environ, cwd=temp_dir,
-                               stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                               stderr=subprocess.PIPE)
             p.check_returncode()
             if p.stderr:
-                # if no device code exists for sm_XX, nvprune exits normally
-                # with a warning printed to stderr
-                assert b'No device code' in p.stderr
-                cufft_lib_pruned = None
-                _cufft_static_no_device_code.append(arch)
+                # maybe this is a new arch that we haven't seen or unsupported
+                # by the current CUDA ver?
+                err = f'nvprune failed with sm_{arch}:\n{p.stderr}'
+                raise RuntimeError(err)
             else:
                 # atomic move with the destination guaranteed to be overwritten
                 # (using os.replace() is also ok here)

